docs/gatsby-node.js

```diff
@@ -35,3 +37,65 @@ exports.onCreateWebpackConfig = ({actions, plugins, loaders, getConfig}) => {
 
   actions.replaceWebpackConfig(config)
 }
+
+exports.sourceNodes = ({actions, createNodeId, createContentDigest}) => {
+  const {createNode} = actions
+
+  // Extract compontent metadata from source files
+  const files = globby.sync(['../src/**/*.tsx'], {absolute: true})
+  const data = docgen.parse(files, {
+    savePropValueAsString: true,
+    propFilter: prop => {
+      if (prop.declarations !== undefined && prop.declarations.length > 0) {
+        const hasPropAdditionalDescription = prop.declarations.find(declaration => {
+          return !declaration.fileName.includes('node_modules')
+        })
+
+        return Boolean(hasPropAdditionalDescription)
+      }
+
+      return true
+    }
+  })
+
+  const components = data.map(component => {
+    return {
+      name: component.displayName,
```

Comment:
Does every component have a `displayName`?

Reply:
Yes, according to the `react-docgen-typescript` types: https://github.com/styleguidist/react-docgen-typescript/blob/6c2a088a2ed86dc8f7b23abea50f13ff78825d76/src/parser.ts#L22